File: part-4/week-1/FloydWarshall/Edge.java
class Edge {
  private int dest;
  private int weight;

  Edge(int dest, int weight) {
    this.dest = dest;
    this.weight = weight;
  }

  int getDest() {
    return this.dest;
  }

  int getWeight() {
    return this.weight;
  }
}

File: part-4/week-1/johnsons_algorithm/Edge.cc
#include "Edge.hh"

Edge::Edge(int __dest, int __weight)
    : dest(__dest)
    , weight(__weight)
{
}

int Edge::get_dest(void)
{
    return this->dest;
}

int Edge::get_weight(void)
{
    return this->weight;
}

void Edge::update_weight(int __weight)
{
    this->weight = __weight;
}